test/cpp/api/functional.cpp

```diff
@@ -1162,8 +1162,16 @@ TEST_F(FunctionalTest, SoftplusDefaultOptions) {
   ASSERT_TRUE(torch::allclose(y, y_exp));
 }
 
+TEST_F(FunctionalTest, Fold) {
+  auto input = torch::randn({1, 3 * 2 * 2, 12});
+  auto output = F::fold(input, FoldOptions({4, 5}, {2, 2}));
+  auto expected_sizes = std::vector<int64_t>({1, 3, 4, 5});
+
+  ASSERT_EQ(output.sizes(), expected_sizes);
```

Comment:
It would be awesome to have an expected value test for `output` as well :D

Reply:
Sure thing!
If you'd like, I'd be happy to add an expected value test for the `output` under `nn::Unfold`/`functional::unfold`.

Reply:
It would be more than awesome to have an expected value test for the `output` under `nn::Unfold`/`functional::unfold` as well. Thanks so much for your help! :D